src/reader_history_search.rs

```diff
@@ -235,9 +238,15 @@ impl ReaderHistorySearch {
                 }
             }
 
-            // Make sure tokens are added in reverse order. See #5150
-            for tok in local_tokens.into_iter().rev() {
-                self.add_if_new(tok);
+            if self.mode == SearchMode::LastToken {
+                if let Some(tok) = local_tokens.into_iter().next_back() {
+                    self.add_if_new(tok);
+                }
+            } else {
+                // Make sure tokens are added in reverse order. See #5150
+                for tok in local_tokens.into_iter().rev() {
+                    self.add_if_new(tok);
```

Comment:
wouldn't it be simpler to add `if self.mode == SearchMode::LastToken { break }` here instead

Reply:
Indeed. I initially used `break`, but it didn't work due to another part. Then I used this form to verify something and forgot to change it back. Now updated.